torch/distributed/optim/utils.py

```diff
@@ -24,6 +24,20 @@
     optim.Adamax: _FunctionalAdamax,
 }
 
+
+# Interface to insert a new functional optimizer to functional_optim_map
+# fn_optim_key and fn_optimizer are user defined
+# Example::
+#     >>> import the new functional optimizer
+#     >>> from xyz import fn_optimizer
+#     >>> from torch.distributed.optim.utils import register_functional_optim
+#     >>> fn_optim_key ="XYZ_optim"
+#     >>> register_functional_optim(fn_optim_key, fn_optimizer)
+
+def register_functional_optim(key, optim):
+    if key not in functional_optim_map:
+        functional_optim_map[key] = optim
```

Comment:
I guess this makes it so that `functional_optim_map` can have torch.optim.Optimizer keys or str keys, but that is fine.